cpp/src/url_test.cpp

```diff
@@ -111,6 +116,54 @@ TEST_CASE("parse URL","[url]") {
                   "amqp", "user", "pass", "::1", "1234", "path",
                   "amqp://user:pass@[::1]:1234/path");
     }
+    SECTION("port") {
+        CHECK("host:1234" == url("amqp://host:1234/path").host_port());
+        CHECK(5672 == url("amqp://foo/path").port_int());
+        CHECK(5671 == url("amqps://foo/path").port_int());
+        CHECK(1234 == url("amqps://foo:1234/path").port_int());
+
+        url u("amqps://foo:xyz/path");
+        url_error caught_error("");
+        try{
+          u.port_int();
+        }
+        catch(url_error& err){
+          caught_error=err;
```

Comment:
It is helpful to have some autoformatter for source code. Either in your IDE, or some cli tool (like clang-format). If you spend the time to configure it to match formatting of the existing code in the project, it will save you time otherwise spent on chasing spaces around equal signs.

Reply:
Thanks, Justin and Jiri for your direction. Jiri, I have installed clang-format but I am not able to configure it to match the formatting of the existing code. Can you please guide me on how to configure it according to this project?

Reply:
Configuring clang-format would be a whole task all of its own ;( The last attempt to come up with a clang-format config for qpid-proton failed, https://github.com/apache/qpid-proton/pull/169. Sorry, this is not what you'd call actionable advice from me.

I have a feature in IDE which runs formatter only on changed lines. Then it usually does not matter that the style I set does not match the rest of the file. For cli, there is git-clang-format which should be able to do that. https://github.com/llvm-mirror/clang/blob/master/tools/clang-format/git-clang-format

Reply:
I have installed git-clang-format and now it's working on cli. Thanks for sharing!!